openedx/core/djangoapps/content_tagging/handlers.py

```diff
@@ -74,3 +89,39 @@ def delete_tag_xblock(**kwargs):
         delete_course_tags.delay(str(xblock_info.usage_key.course_key))
 
     delete_xblock_tags.delay(str(xblock_info.usage_key))
+
+
+@receiver(LIBRARY_BLOCK_CREATED)
+@receiver(LIBRARY_BLOCK_UPDATED)
+def auto_tag_library_block(**kwargs):
+    """
+    Automatically tag Library Blocks based on metadata
+    """
+    library_block_data = kwargs.get("library_block", None)
+    if not library_block_data or not isinstance(library_block_data, LibraryBlockData):
+        log.error("Received null or incorrect data for event")
+        return
+
+    if not CONTENT_TAGGING_AUTO.is_enabled():
+        return
+
+    current_request = crum.get_current_request()
+    update_library_block_tags.delay(
+        str(library_block_data.usage_key), current_request.LANGUAGE_CODE
+    )
+
+
+@receiver(LIBRARY_BLOCK_DELETED)
+def delete_tag_library_block(**kwargs):
+    """
+    Automatically delete Library Blocks auto tags.
+    """
+    library_block_data = kwargs.get("library_block", None)
+    if not library_block_data or not isinstance(library_block_data, LibraryBlockData):
+        log.error("Received null or incorrect data for event")
+        return
+
+    if not CONTENT_TAGGING_AUTO.is_enabled():
+        return
+
+    delete_library_block_tags.delay(str(library_block_data.usage_key))
```

Comment:
Does this really need to be async? I would think that queuing the deletion would use the DB just as much as actually deleting the `ObjectTag`.

I'm tempted to just put `api.delete_object_tags(str(library_block_data.usage_key))` here, maybe in a `try...except` since it's OK if it fails.

Reply:
Agreed, I changed it to be sync code.